src/detect-content.c

```diff
@@ -553,7 +553,11 @@ void DetectContentPropagateLimits(Signature *s)
                             (cd->flags & (DETECT_CONTENT_DEPTH|DETECT_CONTENT_OFFSET|DETECT_CONTENT_WITHIN|DETECT_CONTENT_DISTANCE)) == (DETECT_CONTENT_DISTANCE)) {
                         if (cd->distance >= 0) {
                             // only distance
-                            offset = cd->offset = offset_plus_pat + cd->distance;
+                            if ((uint32_t)offset_plus_pat + cd->distance <= UINT16_MAX) {
```

Comment:
nit: suggest parens around the operands for clarity.

Even though `<=` is lower precedence, it may be clearer.

Reply:
Do you mean `((uint32_t)offset_plus_pat) + ` ?